### ISSUE TITLE
iOS app crashes on picking many videos - iPhone 5s

### ISSUE THREAD

**Issue by ron3001** [miguelpruivo/flutter_file_picker#606]
**Describe the bug**
iOS app crashes on picking many videos. (iPhone 5s only)
**Platform**
iOS
**Platform OS version**
iOS 12.5.1
**How are you picking?**
```
FilePickerResult paths;
    try {
      paths = await FilePicker.platform.pickFiles(
        type: isMedias ? FileType.media : FileType.any,
        allowMultiple: true,
        withData: false,
      );
    } on PlatformException catch (e) {
      print("Unsupported operation" + e.toString());
      throw Exception();
    } catch (ex) {
      print(ex);
      throw Exception();
    }
```
**Details to reproduce the issue**
Attempt to pick files from type media. When Photos opens, choose some videos - total size around 150mb. When the videos are selected, after you choose Select, a spinner shows and spins for around 3 minutes, then the app crashes. Available storage on the device is ~1GB. The spinner shows every time for around 3 minutes, but the crash sometimes does not occur.
**Flutter Version details**
[✓] Flutter (Channel stable, 1.22.0, on Mac OS X 10.15.7 19H114)
• Flutter version 1.22.0
• Dart version 2.10.0

**Comment by miguelpruivo** [comment]
Hi. Try checking the crash log by running it directly with Xcode.

**Comment by ron3001** [comment]
Hi. I run it directly with Xcode.
The error is in FilePickerPlugin.m - line 266 - [paths addObject:asset.localTemporaryPath.absoluteURL];
The exception message is "*** -[__NSArrayM insertObject:atIndex:]: object cannot be nil"

**Comment by miguelpruivo** [comment]
@ron3001 looks that for some reason the file couldn't be cached, hence the NPE. Try freeing a bit more space on the device or pick other files and see if it still happens.

**Comment by ron3001** [comment]
I freed some more space and the issue appears to be gone! Is there some way to catch this specific case?

**Comment by miguelpruivo** [comment]
@ron3001 yes. Actually if I add an error handling for this on native code it might catch the error and throw it to Flutter so you can actually catch it.

I’ll see what I can do.

**Comment by miguelpruivo** [comment]
Fixed with 2.1.7. From now on, the crash shouldn't happen since null check was added there.

Thank you for reporting.
